core/src/main/java/com/datastax/oss/driver/api/core/config/DriverConfigLoader.java

```diff
@@ -49,6 +49,7 @@
    *     changes yet. If this loader does not support programmatic reloading &mdash; which you can
    *     check by calling {@link #supportsReloading()} before this method &mdash; the stage will
    *     fail immediately with an {@link UnsupportedOperationException}.
+   * @throws UnsupportedOperationException if this loader does not support programmatic reloading.
```

Comment:
My idea was to returned a failed `CompletionStage`, not throw the exception directly.